src/odemis/util/graph.py

```diff
@@ -172,6 +172,27 @@ def iter_edges(self, directed: Optional[bool] = None) -> Iterator[Edge]:
                     continue
                 yield (vertex, neighbor)
 
+    def size(self, directed: Optional[bool] = None) -> int:
+        """
+        Return the number of edges in the graph.
+
+        Parameters
+        ----------
+        directed : bool, optional
+            If `True` the edges `(j, i)` and `(i, j)` will be considered as
+            being two separate edges. If `False`, these are counted as a single
+            edge. If not specified, uses `self.directed`.
+
+        Returns
+        -------
+        size : int
+            The number of edges in the graph.
+
+        """
+        counter = itertools.count()
+        collections.deque(zip(self.iter_edges(directed), counter), maxlen=0)
+        return next(counter)
```

Comment:
I think it's simpler to write:
```suggestion
        return sum(1 for _ in self.iter_edges(directed))
```

(and I think I found the stackoverflow article where it stated the deque option was a tiny bit faster, but that's not even correct for Python 3 anymore. `sum()` is now both simpler to read and faster)

Reply:
I agree that Érics suggestion is a lot more readable

Reply:
done